bids2table/entities.py

```diff
@@ -257,7 +222,34 @@ def with_update(
             data.update(entitities)
         if kwargs:
             data.update(kwargs)
-        return BIDSEntities.from_dict(data)
+        return self.__class__.from_dict(data)
+
+
+def make_bids_field(
+    entity_schema: Dict[str, Any],
+) -> Tuple[str, Any, Any]:
+    """
+    BIDS entity dataclass field.
+    """
+    metadata = {
+        "name": entity_schema["name"],
+        "display_name": entity_schema["display_name"],
+        "allowed_values": entity_schema.get("enum"),
+    }
+    type_ = {"integer": int}.get(entity_schema["type"], str)
```

Comment:
So they're all strings, but `format: index` indicates that they should be interpreted as integers:

```suggestion
    type_ = {"index": int}.get(entity_schema["format"], str)
```